src/main/java/com/marek/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -36,4 +36,19 @@ public BookDto findById(Long id) {
                 bookRepository.findById(id).orElseThrow(
                         () -> new EntityNotFoundException("Can't find book with id: " + id)));
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto createBookRequestDto) {
+        Book model = bookMapping.toModel(createBookRequestDto);
+        model.setId(id);
+        bookRepository.updateBookById(id, model);
+        return bookMapping.toBookDto(model);
+    }
```

Comment:
Can be done like this instead:

```suggestion
    @Override
    public BookDto updateById(Long id, CreateBookRequestDto book) {
        Book modelBook = bookMapper.toModel(book);
        modelBook.setId(id);
        return bookMapper.toDto(bookRepository.save(modelBook));
    }
```

Reply:
Pls implement as suggested. The code is more readable and concise that way.